toolchains/gcc_arm_none_eabi/BUILD

```diff
@@ -31,7 +31,10 @@ package(default_visibility = ["//visibility:public"])
 
 _DEVICE_CONFIGS = CORTEX_M_DEVICE_CONFIGS
 
-compiler_components()
+compiler_components(
+    injected_hdr_deps = "@bazel_embedded_upstream_toolchain//:injected_headers",
+    system_hdr_deps = "@bazel_embedded_upstream_toolchain//:polyfill",
```

Comment:
Is the idea that users of `bazel_embedded` can import an upstream repo with `polyfill` and/or ` injected_headers` and must name it `bazel_embedded_upstream_toolchain` for this to work?

Reply:
yep, your facade rules are much more elegant but using them without a copy pasta will introduce a circular dependency. At the moment though this will do the job. If your facades ever end up in skylib I will probably change this. About to push a patch to the pigweed side.

Reply:
Cool. Sounds good to me! Yeah, this solution seems straightforward. We might want to add some docs somewhere to outline that expected usage (maybe a docstring for `compiler_components`?), but totally up to you 😄 Thanks!

Reply:
Yeah of course. There is a lot of stuff that is lacking documentation. Definitely need to up my game in that department! I'll get on to that in the next week or so.

Reply:
I'm probably going to go through everything first and make a few things in the .bzl libraries private, and then spend a bit of time trying to make the public interface really well documented. For example it doesn't neccsarily make sense for compiler_components to be public in the first place. Ideally I will document both public/private stuff a bit better but I'll probably just chip away at that as I go.